benchmarks/_benchmark.py

```diff
@@ -47,9 +48,27 @@ def arrow_info() -> Dict[str, Any]:
     }
 
 
+@dataclass
+class BenchmarkResult:
+    name: str
+    run_id: str = None
+    batch_id: str = None
+    result: Dict = None
+    # `params` (i.e. case) is currently a subset of `tags`, but we often need it separately
+    params: Dict[str, Any] = None
+    tags: Dict[str, str] = None
+    info: Dict[str, Any] = None
+    context: Dict[str, Any] = None
+    github: Dict[str, Any] = None
+    options: Dict[str, Any] = None
+    output: str = None
+    error: str = None
```

Comment:
So I took a bit of a different approach from arrowbench for this one, to a large extent because Python objects are much easier to consistently turn to JSON when you need. So this class is very much just a data class, without anything beyond basic dunder methods. 

The basic contract for now is just that `Benchmark`'s methods that return results will return an instance of this class, regardless of success or failure or whatever. In the future, I suspect we'll call `dataclasses.asdict()` on these and write them to JSON files, which then our separate tool can pick up and send to conbench. That tool may want a richer version of this class with more methods, but I don't want to jump the gun yet. It may very well make sense to put this class in a dep package with all the running code which is currently in {conbench} that both can use.

And good call, I should add a docstring; I need to go figure out what the preferred way of documenting dataclasses is. For now I didn't make sub-objects (like a `Github` class that defines the structure of what goes in `github`) because I suspect a lot of that may change thoughout the process, but eventually that's a better way of really defining this schema than text documentation.

Reply:
Added a big docstring for now. [Numpy style](https://numpydoc.readthedocs.io/en/latest/format.html) (specified in the "How to Python at Voltron" notion page) is vague on how to document individual fields, so I did it at the class level; we'll have to be careful to keep it synced.